cmd/run.go

```diff
@@ -40,3 +55,87 @@ func (h *Handler) Run(ctx context.Context, req *entity.CommandRequest) error {
 
 	return nil
 }
+
+func (h *Handler) runInDocker(ctx context.Context, pwd string, envs *entity.Envs) error {
+	// Start building the image
+	projectCfg, err := h.cfg.GetProjectConfigs()
+	if err != nil {
+		return err
+	}
+
+	project, err := h.ctrl.GetProject(ctx, projectCfg.Project)
+	if err != nil {
+		return err
+	}
+
+	// Strip characters not allowed in Docker image names
+	sanitiser := regexp.MustCompile(`[^A-Za-z0-9_-]`)
+	imageName := sanitiser.ReplaceAllString(project.Name, "")
+	image := fmt.Sprintf("railway-local/%s:latest", imageName)
+
+	buildArgs := []string{"build", "-q", "-t", image, pwd}
+
+	// Build up env
+	for k, v := range *envs {
+		buildArgs = append(buildArgs, "--build-arg", fmt.Sprintf("%s=\"%+v\"", k, v))
+	}
+
+	buildCmd := exec.Command("docker", buildArgs...)
+	ui.StartSpinner(&ui.SpinnerCfg{
+		Message: fmt.Sprintf("Building %s from Dockerfile...", image),
+		Tokens:  ui.TrainEmojis,
+	})
+
+	out, err := buildCmd.CombinedOutput()
+	if err != nil {
+		ui.StopSpinner("")
+		return showCmdError(buildCmd.Args, out, err)
+	}
+
+	ui.StopSpinner(fmt.Sprintf("🎉 Built %s", image))
+
+	port := 4411
```

Comment:
Rng port vs 3000 vs 4411? Can see all of em with pros and cons

Reply:
I'd probably go RNG if I had to pick again; 4411 was picked because it gets pretty close to looking like "RAIL"